aten/src/ATen/core/ivalue.h

```diff
@@ -1281,7 +1281,7 @@ struct TORCH_API IValue final {
             0;
 
     TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
-        uint32_t(tag) >= 0 && uint32_t(tag) < kNumTags,
+        uint32_t(tag) && uint32_t(tag) < kNumTags,
```

Comment:
I think this is slightly not equivalent to the previous code.
`uint32_t(tag) >= 0 ` accepted the value `tag == 0` and the new condition `uint32_t(tag) &&` seems to include it, right?

If this case is not needed, should `uint32_t(tag) && ` be dropped as well from the check?

Reply:
Or if really a check was meant, it should be instead `tag >= 0`...

Reply:
yeah you're right